mlflow/protos/service.proto

```diff
@@ -682,9 +682,21 @@ message ListExperiments {
   // If unspecified, return only active experiments.
   optional ViewType view_type = 1;
 
+  // Maximum number of experiments desired. If specified, max_results can be at most 10,000. 
+  // Servers may impose a default max_results of their choice if unspecified by the client. 
+  // Callers of this endpoint are encouraged to pass max_results explicitly and leverage page_token 
+  // to iterate through experiments.
+  optional int64 max_results = 2 [default = 10000];
```

Comment:
Hi @wamartin-aml. After some offline design discussions, the largest `max_results` value that can be accommodated by Databricks is currently 1,000 (rather than 10,000). AML is free to set a higher threshold. If possible, can we update the default value and documentation here to reflect 1,000 as the default and the maximum threshold that all backends must support?

cc @harupy